src/layer/size.cpp

```diff
@@ -0,0 +1,36 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "size.h"
+
+namespace ncnn {
+
+Size::Size()
+{
+    one_blob_only = true;
+    support_inplace = false;
+}
+
+int Size::forward(const Mat& bottom_blob, Mat& top_blob, const Option& opt) const
+{
+    top_blob.create(1, 4u, opt.blob_allocator);
+    if (top_blob.empty())
+        return -100;
+
+    top_blob[0] = bottom_blob.w * bottom_blob.h * bottom_blob.d * bottom_blob.c;
```

Comment:
onnx size outputs an integer tensor
The code implemented here by ncnn size will return a float tensor
For very large numbers, there is a loss of precision when using float
Will this cause difficulty in subsequent layer processing? For example, perform other addition, subtraction or modulo division operations on the result of size.

Reply:
for things like rms norm etc, it might have a little impact. but, the issue is that the size of batches (which ncnn does not have by itself), will have this. other than devisions for normalizations, I dont think it has other impacts. but if necessary (which is better), I might change the types from float to int. should anything else be done to make it compatible with integers?

Reply:
I'm worried that your original intention was to return integer, but you accidentally returned float tensor. Now it seems that I have unnecessary worries.  :D

Regarding batch, the process of converting the model to ncnn will require that the batch is always 1. There may be special cases, but that will also be solved in the converter, such as intepret nchw to cdhw in ncnn as a workaround. The layer itself doesn't care about it